tests/conformance/common.go

```diff
@@ -0,0 +1,206 @@
+package conformance
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"fortio.org/fortio/log"
+	"github.com/dapr/components-contrib/pubsub"
+	p_redis "github.com/dapr/components-contrib/pubsub/redis"
+	"github.com/dapr/components-contrib/secretstores"
+	ss_local_env "github.com/dapr/components-contrib/secretstores/local/env"
+	ss_local_file "github.com/dapr/components-contrib/secretstores/local/file"
+	"github.com/dapr/components-contrib/state"
+	s_redis "github.com/dapr/components-contrib/state/redis"
+	conf_pubsub "github.com/dapr/components-contrib/tests/conformance/pubsub"
+	conf_secret "github.com/dapr/components-contrib/tests/conformance/secretstores"
+	conf_state "github.com/dapr/components-contrib/tests/conformance/state"
+	"github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	"github.com/dapr/dapr/pkg/components"
+	config "github.com/dapr/dapr/pkg/config/modes"
+
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+)
+
+// nolint:gochecknoglobals
+var testLogger = logger.NewLogger("testLogger")
+
+type TestConfiguration struct {
+	ComponentType string          `yaml:"componentType,omitempty"`
+	Components    []TestComponent `yaml:"components,omitempty"`
+}
+type TestComponent struct {
+	Component     string            `yaml:"component,omitempty"`
+	AllOperations bool              `yaml:"allOperations,omitempty"`
+	Operations    []string          `yaml:"operations,omitempty"`
+	Config        map[string]string `yaml:"config,omitempty"`
+}
```

Comment:
Nit: Personally, I don't like the redundant info from AllOperations and Operations. In tests like this, I like to be really prescriptive and only have the operations that are being tested as it gives you more information in a single point instead of having to cross-reference this with the tests.

I'm fine with it as is though, just thought I'd bring up the opinion.

Reply:
can be changed with the next PR so that if no operations are specified but the component is specified it defaults to all operations if not only the specified ops are run.